test/core/pthread/test_pthread_keepalive.c

```diff
@@ -0,0 +1,63 @@
+#include <assert.h>
+#include <stdatomic.h>
+#include <stdbool.h>
+#include <pthread.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include <emscripten/emscripten.h>
+#include <emscripten/eventloop.h>
+#include <emscripten/console.h>
+#include <emscripten/proxying.h>
+
+pthread_t t;
+pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+
+void timeout(void* arg) {
+  _emscripten_errf("timeout");
+  pthread_mutex_lock(&mutex);
+  pthread_cond_signal(&cond);
+  pthread_mutex_unlock(&mutex);
```

Comment:
This needs to set a flag protected by the lock so the waiter doesn't start waiting after the signal has already been sent. Also it would be more efficient to release the lock before calling `pthread_cond_signal`.